mpcs/src/util/merkle_tree.rs

```diff
@@ -23,36 +23,89 @@ use ark_std::{end_timer, start_timer};
 use super::hash::write_digest_to_transcript;
 
 #[derive(Clone, Debug, Default, Serialize, Deserialize)]
-#[serde(bound(serialize = "E: Serialize", deserialize = "E: DeserializeOwned"))]
-pub struct MerkleTree<E: ExtensionField>
+#[serde(bound(deserialize = "E: DeserializeOwned"))]
+pub struct MerkleTreeDigests<E: ExtensionField>
 where
     E::BaseField: Serialize + DeserializeOwned,
 {
     inner: Vec<Vec<Digest<E::BaseField>>>,
```

Comment:
You might want to leave a doc comment on the struct explaining how this vector encodes a tree?